### PR TITLE
K8SPXC-264 Wait for patch propagation

### PR DESCRIPTION
**Pull request by cap1984**

### CODE REVIEW

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster-operator?pullRequest=382) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster-operator?pullRequest=382) before we can accept your contribution.<br/><hr/>**Iv Pil** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-xtradb-cluster-operator?pullRequest=382) it.</sub>

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| storage | passed |
| scaling | passed |
| init-deploy | passed |
| recreate | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| limits | passed |
| monitoring | failed |
| restore-to-encrypted-cluster | failed |
| upgrade | failed |
| self-healing-advanced | failed |
| operator-self-healing | passed |
| demand-backup | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| recreate | passed |
| big-data | passed |
| storage | passed |
| scaling | failed |
| scaling-proxysql | failed |
| self-healing | passed |
| limits | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| self-healing-advanced | passed |
| security-context | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | failed |
| auto-tuning | passed |
| scheduled-backup | failed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| big-data | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| scaling | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| demand-backup-encrypted-with-tls | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| affinity | passed |
| scaling-proxysql | passed |
| one-pod | passed |
| upgrade | passed |
| big-data | passed |
| auto-tuning | passed |
| storage | passed |
| upgrade-consistency | passed |
| proxysql-sidecar-res-limits | passed |
| self-healing | passed |
| security-context | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | passed |

File: e2e-tests/upgrade-consistency/run [hunk 1]
```diff
@@ -8,6 +8,27 @@ test_dir=$(realpath $(dirname $0))
 
 cluster='some-name'
 
+wait_for_sts_generation() {
+    local name="${1}"
+    local desired_generation="${2}"
+    local size="${3}"
+    set +o xtrace
+    retry=0
+    until [[ "$(kubectl_bin get sts $name -o jsonpath='{.status.observedGeneration}')" == "${desired_generation}" \
```
Comment (delgod) [inline e2e-tests/upgrade-consistency/run:17]: can we add addition check on CR.status.observedGeneration?
to make sure that operator procesed new CR

Comment (cap1984) [reply e2e-tests/upgrade-consistency/run:17]: sure

Comment (cap1984) [reply e2e-tests/upgrade-consistency/run:17]: Sorry, in this very test generation changes only once. So it's not a relevant check.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| init-deploy | passed |
| scaling | passed |
| big-data | failed |
| storage | failed |
| scaling-proxysql | failed |
| limits | passed |
| monitoring | failed |
| monitoring-2-0 | failed |
| affinity | passed |
| restore-to-encrypted-cluster | failed |
| upgrade | passed |
| demand-backup | passed |
| self-healing | passed |
| upgrade-consistency | passed |
| one-pod | failed |
| auto-tuning | failed |
| security-context | passed |
| proxysql-sidecar-res-limits | passed |
| self-healing-advanced | failed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | failed |
| operator-self-healing | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| big-data | failed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | failed |
| monitoring-2-0 | failed |
| scaling | passed |
| scaling-proxysql | failed |
| self-healing | passed |
| self-healing-advanced | failed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| affinity | passed |
| one-pod | passed |
| operator-self-healing | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| storage | passed |
| big-data | failed |
| scaling | passed |
| init-deploy | passed |
| limits | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | failed |
| upgrade | passed |
| monitoring | failed |
| demand-backup | failed |
| monitoring-2-0 | failed |
| self-healing-advanced | failed |
| affinity | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| security-context | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| limits | passed |
| security-context | passed |
| monitoring | failed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | failed |
| monitoring-2-0 | failed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| demand-backup | failed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| recreate | passed |
| storage | passed |
| restore-to-encrypted-cluster | passed |
| self-healing | passed |
| demand-backup | passed |
| self-healing-advanced | failed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | failed |
| monitoring-2-0 | failed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| monitoring | failed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| storage | passed |
| scheduled-backup | passed |
| scaling | passed |
| self-healing | passed |
| demand-backup-encrypted-with-tls | passed |
| scaling-proxysql | passed |
| self-healing-advanced | failed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| big-data | passed |
| monitoring-2-0 | failed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| scaling | passed |
| init-deploy | passed |
| self-healing | passed |
| recreate | passed |
| scaling-proxysql | passed |
| limits | passed |
| self-healing-advanced | failed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| monitoring | failed |
| demand-backup | passed |
| upgrade-consistency | passed |
| scheduled-backup | passed |
| security-context | passed |
| demand-backup-encrypted-with-tls | passed |
| big-data | passed |
| monitoring-2-0 | failed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| recreate | passed |
| big-data | passed |
| upgrade-consistency | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| demand-backup | passed |
| init-deploy | passed |
| scheduled-backup | passed |
| limits | passed |
| demand-backup-encrypted-with-tls | passed |
| monitoring | failed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | failed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| init-deploy | passed |
| storage | passed |
| limits | passed |
| self-healing | passed |
| monitoring | failed |
| self-healing-advanced | passed |
| recreate | passed |
| operator-self-healing | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scaling | passed |
| scheduled-backup | passed |
| scaling-proxysql | passed |
| demand-backup-encrypted-with-tls | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| monitoring-2-0 | failed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |

**Pull request merged**